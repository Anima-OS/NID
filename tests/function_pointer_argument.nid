// function pointer as argument
int set_callback(int (*callback)(int a, int b));