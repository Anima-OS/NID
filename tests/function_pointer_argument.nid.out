{ declarations: 
   [ { function: 
        { type: { name: 'int' },
          name: 'set_callback',
          attributes: {},
          parameters: 
           [ { type: 
                { name: 'int',
                  parameters: 
                   [ { type: { name: 'int' }, name: 'a', attributes: {} },
                     { type: { name: 'int' }, name: 'b', attributes: {} } ],
                  functionPointer: true },
               attributes: {},
               name: 'callback' } ] } } ] }
